state/src/storage/cursor.rs

```diff
@@ -396,4 +427,46 @@ mod test {
         assert_eq!(k, expected.to_string());
         assert_eq!(v, expected);
     }
+
+    /// Tests a filtered cursor
+    #[test]
+    fn test_filtered_cursor() {
+        // Create a db and insert 0..N into the table randomly
+        const N: usize = 1000;
+        let db = mock_db();
+        db.create_table(TEST_TABLE).unwrap();
+
+        insert_n_random(&db, N);
+
+        // Create a filtered cursor that only returns even keys
+        let tx = db.new_raw_read_tx().unwrap();
```

Comment:
the db auto-orders string keys, hence using a string key and not a numeric type.